### PR TITLE
[Next Docs] Update Git Workflow

### PR DESCRIPTION
**Pull request by delbaoliveira**
Update our git workflow in preparation for open-sourcing the content of the docs ([linear task](https://linear.app/vercel/issue/DX-1579/set-up-github-workflow)).

**Templates:**
- [x] Update docs issue template to encourage contributions
- [x] Update PR template to include link to new contribution guide 

**Code Owners / Reviewers:** 
- https://github.com/vercel/next.js/pull/50841

**Labels:** 
- [x] Add DevEx team to labeler.json so PRs get the "created by: DevEx team"

**Other:** 
- [x] Remove docs manifest from CI checks as we no longer have one (keep the manifest for errors as they live under `/pages`)
- [x] Add `unifiedjs.vscode-mdx` to the vscode extension list

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: fcf82a0def94e763911af7cb3af4a38bf4b6a7b4

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by delbaoliveira** [comment]
To do: add `unifiedjs.vscode-mdx` to the [vscode extensions list](https://github.com/vercel/next.js/blob/canary/.vscode/extensions.json)

**Review by manovotny — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/4.docs_request.yml [hunk 1]
```diff
@@ -4,6 +4,12 @@ title: 'Docs: '
 labels:
   - 'template: documentation'
 body:
+  - type: markdown
+    attributes:
+      value: Before opening this issue to request a docs improvement, is this something you can do yourself? Your contributions are welcomed and appreciated. See our [Docs Contribution Guide](https://nextjs.org/docs/community/contribution-guide) to learn how to edit the Next.js docs.
```
[the remark this answers is not in the document]
Comment (delbaoliveira) [reply .github/ISSUE_TEMPLATE/4.docs_request.yml:9]: Much better!

Comment (delbaoliveira) [inline .github/ISSUE_TEMPLATE/4.docs_request.yml:9]:
```suggestion
      value: Before opening this issue to request a docs improvement, is this something you can help us with? Your contributions are welcomed and appreciated. See our [Docs Contribution Guide](https://nextjs.org/docs/community/contribution-guide) to learn how to edit the Next.js docs.
```

**Review by manovotny — DISMISSED** [review]

**Review by manovotny — APPROVED** [review]

**Review by leerob — APPROVED** [review]

**Review by huozhi — APPROVED** [review]

**Review by huozhi — APPROVED** [review]

**Pull request merged**